akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -320,20 +304,74 @@ class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[Sour
     runWith(Sink.fold(zero, f), materializer)
 
   /**
-   * Concatenates a second source so that the first element
-   * emitted by that source is emitted after the last element of this
-   * source.
+   * Concatenate the second [[Source]] to current one, meaning that once current
+   * is exhausted and all result elements have been generated,
+   * the second Source’s elements will be produced.
+   */
+  def concat[T >: Out, M](second: Graph[SourceShape[T], M]): javadsl.Source[T, Mat] =
```

Comment:
@agolubev The Scala Standard Library calls the parameter `that`, so you can talk about `this` and `that` in the docs. Ref: http://www.scala-lang.org/api/2.10.4/index.html#scala.collection.immutable.List